app/components/bulk_upload_error_row_component.html.erb

```diff
@@ -49,24 +49,24 @@
             <% end %>
           <% end %>
 
-          <%= table.with_body do |body| %>
-            <% potential_errors.group_by(&:error).each do |error_message, errors| %>
-              <% errors.each_with_index do |error, index| %>
-                <% row_class = "grouped-rows" %>
-                <% row_class += " first-row" if index.zero? %>
-                <% row_class += " last-row" if index == errors.size - 1 %>
-                <% body.with_row(html_attributes: { class: row_class }) do |row| %>
-                  <% row.with_cell(text: error.cell) %>
-                  <% row.with_cell(text: question_for_field(error.field), html_attributes: { class: "govuk-!-width-one-half" }) %>
-                  <% if index == 0 %>
-                    <% row.with_cell(text: error_message.html_safe, rowspan: errors.size, html_attributes: { class: "govuk-!-font-weight-bold govuk-!-width-one-half grouped-multirow-cell" }) %>
-                  <% end %>
-                  <% row.with_cell(text: error.field.humanize) %>
+        <%= table.with_body do |body| %>
+          <% potential_errors.group_by(&:error).each_with_index do |(error_message, errors), group_index| %>
+            <% total_groups = potential_errors.group_by(&:error).size %>
+            <% errors.each_with_index do |error, index| %>
+              <% row_class = row_classes(index, errors.size) %>
+              <% body.with_row(html_attributes: { class: row_class }) do |row| %>
+                <% row.with_cell(text: error.cell) %>
+                <% row.with_cell(text: question_for_field(error.field), html_attributes: { class: "govuk-!-width-one-half" }) %>
+                <% if index == 0 %>
+                  <% cell_class = cell_classes(group_index, total_groups) %>
+                  <% row.with_cell(text: error_message.html_safe, rowspan: errors.size, html_attributes: { class: cell_class }) %>
                 <% end %>
+                <% row.with_cell(text: error.field.humanize) %>
```

Comment:
We group errors together to merge the error message cell. In order to remove the bottom border of the last group I needed to know the total number of groups.

For line 62, we add the `grouped-multirow-cell` class to the cell within the first row of the grouped error messages so that we can put back the bottom border just like with the last row in a group. I changed this to not put back the bottom border when the last row in the group was also the last row in the table.